### PR TITLE
List apps based on address, connect to app based on address

### PR DESCRIPTION
**Pull request by david-yan**
Resolves https://github.com/ZondaX/ledger-oasis-go/issues/3

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,4 +9,5 @@ require (
 	github.com/pkg/errors v0.8.1 // indirect
 	github.com/stretchr/testify v1.3.0
 	github.com/zondax/hid v0.9.0 // indirect
+	github.com/zondax/ledger-go v0.9.0
```
Comment (kostko) [inline go.mod]: Since this was changed, the old dependency should probably be removed from `go.mod` and `go.sum`.

File: app.go [hunk 2]
```diff
@@ -45,6 +45,55 @@ type LedgerOasis struct {
...
+func ConnectLedgerOasisApp(seekingAddress string, path []uint32) (*LedgerOasis, error) {
+	for _, d := range ledger_go.FindLedgerDevices() {
+		device, err := d.Open()
+		if err != nil {
+			continue
+		}
+		ledgerAPI := ledger_go.NewLedger(device)
+		app := LedgerOasis{ledgerAPI, VersionInfo{}}
+
+		_, address, err := app.GetAddressPubKeyEd25519(path)
```
Comment (kostko) [inline app.go]: What kind of address are you comparing here? This should be something that is independent of the derivation path and only dependent on the actual device (like a device ID).

Comment (david-yan) [reply app.go]: I responded this in the integration PR, but from conversations with @jleni , there is not reliable device ID, so we are using address to identify devices.

Comment (jleni) [reply app.go]: Correct. Devices themselves do not have a serial number and are not identifiable.

**Comment by jleni** [comment]
mm.. I don't seem to have permission to push my changes to this PR.

**Comment by jleni** [comment]
maybe @david-yan you can give me access to your fork?

**Comment by david-yan** [comment]
@jleni added you as a collaborator.

**Pull request merged**